### ISSUE TITLE
[Feature] add to IOS

### ISSUE THREAD

**Issue by Nauris62** [PhoneVR-Developers/PhoneVR#100]
I would love this on ios couse im a strictly iphone user

**Comment by NintyS** [comment]
If something is needed like remote MacOS to make builds I can help.

**Comment by ShootingKing-AM** [comment]
Thanks for your interest, but last time I checked, to develop iOS apps, we need to pay a lot of money to Apple Inc, to even start development. I personally don't have any iOS devices to test myself. Also I feel if you can afford an iOS device, you can easily invest in a standalone vr hardware (i guess?)

**Comment by NintyS** [comment]
No, developing app is free. Releasing it costs. Free version ( Apple Developer account) of app have 10 days to expire as far as I remember. Paid version have expire date up to one year. Only apps from App Store don't expire. Yes, if you can buy a brand new iPhone then you can buy a standalone device too, but from what I observed in my group most people buy a 2/3 years old iPhone or gets them ( used ) from family.

I have Apple devices and I can try to help develop iOS version ( but I'm not fluent in Switch or Objective-C )

**Comment by ShootingKing-AM** [comment]
IIRC there is a constraint of dev environment too right ? Like need to develop (run the IDE) on MacOS etc.
Even am not fluent in Swift, but once you start shouldn't be hard.

**Comment by NintyS** [comment]
Yea, you need to have Mac with MacOS which supports newest xCode version to program on newest iOS versions.

If you want I can try create iOS client in Qt & C++.

**Comment by ShootingKing-AM** [comment]
Maybe you can try researching software platforms which are available for both Android and iOS, instead of writing in native iOS (Swift etc.). I am aware that there are packages like flutter (etc.) which support multi-os with the same code base. 

Maybe we can somehow integrate flutter (or other) with current code base. Its not a good idea to write the same code in two different languages, since it would be hard to maintain. That said, I myself don't know how to integrate flutter with ndk to port it to iOS, or even if it supports NDK. 

The smartphone market is around 75% Android and 25% iOS IIRC. Hence, Supporting iOS would make us available for even more users. Any PRs are welcome though.

**Comment by NintyS** [comment]
Then Qt supposed to be good. I can create one code and build it for two or more platforms

**Comment by ShootingKing-AM** [comment]
I hope you are talking about mobile, never heard of Qt for android, but still try to reuse the existing codebase as much as possible.

If you are talking about server codebase, SteamVR API(which PVR uses) itself is not supported for linux/macos. In the latest commits to master branch, I made an integration with ALVR, which has some experimental linux support.

**Comment by NintyS** [comment]
Yes, I'm saying about mobile. You can write apps in Qt for iOS, Android And Ubuntu Touch ( but there you need to do a lots of work ). I will try to reuse the code.

**Comment by NintyS** [comment]
@ShootingKing-AM can you throw some informations about how to connect to ALVR server and if google cardboard is required?

**Comment by ShootingKing-AM** [comment]
ALVR has a library that can statically linked via Android NDK, that alvr_client_core can be used to communicate with ALVR Server. Yes Google carboard is required for its API. 

ALVR example: https://github.com/PhoneVR-Developers/PhoneVR/blob/master/code/mobile/android/PhoneVR/app/src/main/cpp/alvr_main.cpp

**Comment by OWL4C** [comment]
Maybe this is a bit presumptuous but why would you need to create an iphone app directly in contrast to streaming the video to a web browser (similar to deskreen or weylus)
There are web apis to use a gyroscope from a browser, so that too should be doable? 
I am not trying to belittle anyone or diminish any work already done, i am genuinely interested and wonder if a completely endpoint independent software would be possible

**Comment by runephilosof** [comment]
You can do free distribution on iOS as well.

https://altstore.io/

**Comment by Julmik647** [comment]
> Maybe this is a bit presumptuous but why would you need to create an iphone app directly in contrast to streaming the video to a web browser (similar to deskreen or weylus) There are web apis to use a gyroscope from a browser, so that too should be doable? I am not trying to belittle anyone or diminish any work already done, i am genuinely interested and wonder if a completely endpoint independent software would be possible

I don't know if we can make it work with webkit limitations. Webkit is currently used by Safari and every browser on iOS.
